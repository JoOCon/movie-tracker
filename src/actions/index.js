export const addMovies = (movieData) => ({
  type: 'ADD_MOVIES',
  movieData
});

export const addFavorite = (movie) => ({
  type: 'ADD_FAVORITE',
  movie
});

export const setActiveUser = ({ id, name }) => ({
  type: 'SET_ACTIVE_USER',
  id,
  name
});

export const setUserFavorites = (favorites) => ({
  type: 'SET_USER_FAVORITES',
  favorites
});