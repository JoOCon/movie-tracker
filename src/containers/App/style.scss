@import '../../styles/variables.scss';

.App {
  text-align: center;
  background-color: $grey;
}
