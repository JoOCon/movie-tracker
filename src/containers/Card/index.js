import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import StarRatings from 'react-star-ratings';
import { addFavorite } from '../../actions';
import { postFavorites } from '../../api/apiCalls';
import './style.css';

export const Card = (props) => {
  const { movie, user, addFavorite } = props;
  const { poster_path, vote_average } = movie;
  const rating = vote_average / 2;

  const handleFavorite = () => {
    const favoriteData = {...movie, user_id: user.id};
    addFavorite(favoriteData);
    postFavorites(favoriteData);
  };

  return (
    <div className='card'>
      <div className='favorite-wrapper'>
        <button 
          className='favorite' 
          onClick={() => handleFavorite()}>
        </button>
        <StarRatings 
          starDimension={'1em'} 
          rating={rating} 
          numberOfStars={5} 
          starRatedColor={'gold'} 
        />
      </div>
      <img className='poster-image' src={poster_path} alt='movie data' />
    </div>
  );
};

const mapStateToProps = (state) => ({
  user: state.userData
});

const mapDispatchToProps = (dispatch) => ({
  addFavorite: (movie) => dispatch(addFavorite(movie)),
  postFavorites: (movie) => dispatch(postFavorites(movie))
});

Card.propTypes = {
  movie: PropTypes.object,
  user: PropTypes.object,
  addFavorite: PropTypes.func,
  postFavorites: PropTypes.func
};

export default connect(mapStateToProps, mapDispatchToProps)(Card);
