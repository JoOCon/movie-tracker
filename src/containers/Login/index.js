import React, { Component } from 'react';
import './style.css';
import { login } from '../../api/apiCalls'

export class Login extends Component {
  constructor() {
    super();
    this.state = {
      email: '',
      password: ''
    };
  }

  handleChange = (e) => {
    e.preventDefault();
    const { name, value } = e.target;
    this.setState({ [name]: value });
  }

  handleSubmit = async (e) => {
    e.preventDefault();
    await login(this.state);
    // this.setState({ userName: '', password: '' });
  }

  render() {
    const { email, password } = this.state;
    return (
      <form className='user-login' onSubmit={this.handleSubmit}>
        <input name='email' placeholder='email' type='email' value={email} onChange={this.handleChange} />
        <input name='password' placeholder='password' type='password' value={password} onChange={this.handleChange} />
        <button>LogIn</button>
      </form>
    );
  }
}
