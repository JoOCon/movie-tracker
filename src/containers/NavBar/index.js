import React from 'react';
import { Link } from 'react-router-dom';
import './style.css';

export const NavBar = () => {
  return (
    <div className='nav-bar'>
      <Link to='/login'><button className='login'>login</button></Link>
      <Link to='/signup'><button className='signup'>SignUp</button></Link>
    </div>
  );
};