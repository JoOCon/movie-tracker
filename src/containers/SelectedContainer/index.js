import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { manageSelectedMovie } from '../../actions';
import './style.css';

export const SelectedContainer = (props) => {
  const { selectedMovie, manageSelectedMovie } = props;
  const {
    overview,
    poster_path,
    release_date,
    title
  } = props.selectedMovie;

  const displayInfo = () => { 
    if (poster_path){
      return (
        <div className='selected-container' onClick={() => manageSelectedMovie(selectedMovie)}>
          <img className='poster-image' src={poster_path} alt='movie data' />
          <div className='selected-movie-content'>
            <h1>{title}</h1>
            <h3>RELEASE: {release_date}</h3>
            <p>DETAILS: {overview}</p>
          </div>
        </div>);
    } else {
      return <div></div>;
    }
  };

  return (
    <div className='selected-wrapper'>{displayInfo()}</div>
  );
};

export const mapStateToProps = (state) => ({
  selectedMovie: state.selectedMovie
});

export const mapDispatchToProps = (dispatch) => ({
  manageSelectedMovie: (movie) => dispatch(manageSelectedMovie(movie))
});

SelectedContainer.propTypes = {
  selectedMovie: PropTypes.object,
  manageSelectedMovie: PropTypes.func
};

export default connect(mapStateToProps, mapDispatchToProps)(SelectedContainer);
