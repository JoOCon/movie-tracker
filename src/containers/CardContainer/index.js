import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Card from '../../components/Card';
import './style.css';

const CardContainer = (props) => {
  const displayMovies = props.movies.map(movie => <Card key={movie.id} movie={movie} />);
  return (
    <div className="card-container">
      {displayMovies}
    </div>
  );
};

const mapStateToProps = (state) => ({
  movies: state.moviesData
});

CardContainer.propTypes = {
  movies: PropTypes.array
};

export default connect(mapStateToProps)(CardContainer);
