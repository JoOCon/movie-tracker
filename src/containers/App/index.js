import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { BrowserRouter, Route } from 'react-router-dom';
import { addMovies } from '../../actions';
import { fetchMovies } from '../../api/apiCalls';
import CardContainer from '../CardContainer';
import FavoriteContainer from '../FavoriteContainer';
import Login from '../../containers/Login';
import SignUp from '../../containers/SignUp';
import { Header } from '../../components/Header';
import './style.css';

export class App extends Component {
  componentDidMount = () => {
    this.fetchOpeningMovies();
  }

  fetchOpeningMovies = async () => {
    const { addMovies } = this.props;
    const openingMovies = await fetchMovies();
    addMovies(openingMovies);
  }

  render() {
    return (
      <BrowserRouter>
        <div className="App">
          <Route path='/' component={Header} />
          <Route exact path='/login' component={Login} />
          <Route exact path='/signup' component={SignUp} />
          <Route exact path='/' component={CardContainer} />
          <Route exact path='/:landingPage' component={CardContainer} />
          <Route exact path='/:user/favorites' component={FavoriteContainer} />
        </div>
      </BrowserRouter>
    );
  }
}

export const mapDispatchToProps = (dispatch) => ({
  addMovies: (moviesData) => dispatch(addMovies(moviesData))
});

App.propTypes = {
  addMovies: PropTypes.func
};

export default connect(null, mapDispatchToProps)(App);
