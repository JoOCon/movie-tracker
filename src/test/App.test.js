import React from 'react';
import { mapDispatchToProps } from '../containers/App';
import { App } from '../containers/App';
import { shallow } from 'enzyme';
import { mockFullData, mockCleanMoviesDataArray, mockCleanMovieData } from '../test/mockData'
import { addMovies } from '../actions'

describe('APP tests', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App addMovies={addMovies} />);
  });

  it('renders the Card with the correct props', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('fetchOpeningMovies', async () => {
    it.skip('should display the opening movies on load', async () => {
      await wrapper.instance().fetchOpeningMovies();

    });
  })

  describe('mapDispatchToProps', () => {
    it.skip('calls dispatch with the addMovies action when fetchOpeningMovies is called', async () => {
      const mockDispatch = jest.fn()
      const mockMovieData = mockCleanMovieData
      const mappedProps = mapDispatchToProps(mockDispatch);
      const expected = mappedProps.addMovies(mockCleanMovieData);
      const result = mockCleanMovieData
      expect(mappedProps).toHaveBeenCalledWith(expected);
    })
  })
});
