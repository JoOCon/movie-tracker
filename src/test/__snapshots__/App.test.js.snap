// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`APP tests renders the Card with the correct props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    addMovies={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="App"
>
        <Route
                component={[Function]}
                path="/"
        />
        <Route
                component={[Function]}
                exact={true}
                path="/login"
        />
        <Route
                component={[Function]}
                exact={true}
                path="/signup"
        />
        <Route
                component={[Function]}
                exact={true}
                path="/"
        />
        <Route
                component={[Function]}
                exact={true}
                path="/:landingPage"
        />
        <Route
                component={[Function]}
                exact={true}
                path="/:user/favorites"
        />
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Route
            component={[Function]}
            path="/"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/login"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/signup"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/:landingPage"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/:user/favorites"
/>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/login",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/signup",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/:landingPage",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/:user/favorites",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="App"
>
          <Route
                    component={[Function]}
                    path="/"
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/login"
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/signup"
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/"
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/:landingPage"
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/:user/favorites"
          />
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              path="/"
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/login"
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/signup"
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/"
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/:landingPage"
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/:user/favorites"
/>,
          ],
          "className": "App",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/login",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/signup",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/:landingPage",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/:user/favorites",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "disableLifecycleMethods": true,
    "hydrateIn": undefined,
  },
}
`;
